src/components/Hits/withClickAnalytics.js

```diff
@@ -0,0 +1,67 @@
+import React, { Component } from 'preact-compat';
+import PropTypes from 'prop-types';
+
+export default C =>
+  class WithClickAnalyticsWrapper extends Component {
+    static propTypes = {
+      hits: PropTypes.array.isRequired,
+      results: PropTypes.object.isRequired,
+    };
+
+    constructor(props) {
+      super(props);
+      this.ref = null;
+    }
+
+    componentDidMount() {
+      const node = this.ref.getDOMNode();
+      node.addEventListener('click', ev => {
+        const event = ev.target.getAttribute('data-analytics-event');
+        const payload = ev.target.getAttribute('data-analytics-payload');
+        if (event && payload) {
+          // event should be oneOf ['clickedObjectIDsAfterSearch', ...]
+          // payload should be valid json
+          let jsonPayload;
+          try {
+            jsonPayload = JSON.parse(payload);
+          } catch (e) {
+            console.error('Invalid JSON event payload');
+            return;
+          }
+
+          // call aa(eventType, eventPayload)
+          console.log(event, jsonPayload);
+        }
+      });
+    }
+
+    render() {
+      const {
+        results: { index, hitsPerPage, page, queryID },
+      } = this.props;
+
+      const hits = this.props.hits.map((hit, position) => ({
+        ...hit,
+        __hitAbsoluteIndex: hitsPerPage * page + position + 1,
+        __index: index,
+        __queryID: queryID,
+        analytics: {
```

Comment:
We should be careful about overriding hits' attributes. We might want to rename it something like `_analytics`?

Reply:
I suggested this as well, but it should be `__analytics` (one for algolia, one for frontend)

Reply:
yes let's make it private